tools/ci_connector_ops/ci_connector_ops/pipelines/actions/tests.py

```diff
@@ -0,0 +1,212 @@
+        stdout=stdout,
+    )
+
+
+async def run_integration_tests(connector_under_test: Container, step=Step.INTEGRATION_TESTS) -> StepStatus:
+    """Run all pytest tests declared in the unit_tests directory of the connector code.
+
+    Args:
+        connector_under_test (Container): The connector under test container.
+        step (Step): The step in which the integration tests are run. Defaults to Step.UNIT_TESTS
+
+    Returns:
+        StepResult: Failure or success of the integration tests with stdout and stdout.
+    """
+    connector_under_test = step.get_dagger_pipeline(connector_under_test)
+    step_status, stderr, stdout = await _run_tests_in_directory(connector_under_test, "integration_tests")
+    return StepResult(
+        step,
+        step_status,
+        stderr=stderr,
+        stdout=stdout,
+    )
+
+
+async def run_acceptance_tests(
+    context: ConnectorTestContext,
+    connector_under_test_image_id: str,
+    step=Step.ACCEPTANCE_TESTS,
+) -> Tuple[StepResult, Directory]:
+    """Runs the acceptance test suite on a connector dev image.
+    It's rebuilding the connector acceptance test image if the tag is :dev.
+
+    Args:
+        context (ConnectorTestContext): The current test context, providing a connector object, a dagger client, a repository directory and the secrets directory.
+        connector_under_test_image_id (str): Connector under test image id, used as a cachebuster.
+        step (Step): The step in which the acceptance tests are run. Defaults to Step.ACCEPTANCE_TESTS
+
+    Returns:
+        Tuple[StepResult, Directory]: Failure or success of the acceptances tests with stdout and stdout AND an updated secrets directory.
+
+    """
+    if not context.connector.acceptance_test_config:
+        return StepResult(Step.ACCEPTANCE_TESTS, StepStatus.SKIPPED), None
+
+    dagger_client = step.get_dagger_pipeline(context.dagger_client)
+
+    docker_host_socket = dagger_client.host().unix_socket("/var/run/docker.sock")
+
+    if context.connector_acceptance_test_image.endswith(":dev"):
+        cat_container = context.connector_acceptance_test_source_dir.docker_build()
+    else:
+        cat_container = dagger_client.container().from_(context.connector_acceptance_test_image)
+
+    cat_container = (
+        cat_container.with_unix_socket("/var/run/docker.sock", docker_host_socket)
+        .with_workdir("/test_input")
+        .with_env_variable("CACHEBUSTER", connector_under_test_image_id)
+        .with_mounted_directory("/test_input", context.get_connector_dir(exclude=["secrets", ".venv"]))
+        .with_directory("/test_input/secrets", context.secrets_dir)
+        .with_entrypoint(["python", "-m", "pytest", "-p", "connector_acceptance_test.plugin", "-r", "fEsx"])
+        .with_exec(["--acceptance-test-config", "/test_input"])
+    )
+
+    secret_dir = cat_container.directory("/test_input/secrets")
+    updated_secrets_dir = None
+    if secret_files := await secret_dir.entries():
```

Comment:
Nit not related to a poc, only if we merge.

Why do you think about:
1. Moving the secret file directory logic to a named function?
2. Move "/test_input/secrets" to a constant